source/api/commands/state.md

```diff
@@ -0,0 +1,148 @@
+---
+title: state
+
+---
+
+Get and set state _synchronously_.
+
+# Syntax
+
+```javascript
+.state()
+.state(key)
+.state(key, value)
+```
+
+## Usage
+
+### Getting
+
+```javascript
+const window = cy.state('window')
+```
+
+```javascript
+const location = cy.state('location')
+```
+
+You can also retrieve all of the existing state synchronously:
+
+```javascript
+const allState = cy.state()
+```
+
+### Setting
+
+```javascript
+cy.state('ids', [1, 2, 3])
+```
+
+The state with the key `ids` can subsequently be accessed synchronously.
+
+```javascript
+const ids = cy.state('ids') // [1, 2, 3]
+```
+
+## Arguments
+
+**{% fa fa-angle-right %} key**  ***(String)***
+
+The `key` of the `value` that is being get or set.
+
+**{% fa fa-angle-right %} value**  ***(any)***
+
+The `value` to be associated with the `key`.
+
+Can be a value of any JavaScript type including `null` and `undefined`.
+
+# Examples
+
+## Get an existing value
+
+***Get the remote window object***
+
+```javascript
+const window = cy.state('window')
+```
+
+This is a _synchronous_ operation. Contrast this with {% url `cy.window()` window %} which _yields_ the remote window object _asynchronously_.
+
+## No Args
+
+***Get all the state***
+
+```javascript
+const allState = cy.state()
+```
+
+This will return all of the existing state. (An `Object`.)
+
+{% note warning %}
+Remember that `.state()` is a _synchronous_ operation. Most Cypress commands are _asynchronous_ meaning they are queued up to be run later. The `.state()` will only reflect any state from those other commands once they have been run.
+{% endnote %}
+
+# Rules
+
+## Timeouts {% helper_icon timeout %}
+
+{% timeouts none .state %}
+
+# Command Log
+
+- `.state()` does *not* log in the Command Log
+
+# Notes
+
+## Primary Use Case
+
+`.state()` is _synchronous_ and as such it's super useful in the implementation of Cypress' core commands and your own {% url "custom commands" custom-commands %} where you can use it to access the remote window object or document and perform actions immediately.
```

Comment:
"it's super useful" -- tone seems slightly off for our docs? Also based on the example, maybe suggest something like "Use `.state()` to store details about a user journey for the duration of a test case, or to perform stateful actions on global objects like window or document."